include/zephyr/sys/time.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2024 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_TIME_H_
+#define ZEPHYR_INCLUDE_SYS_TIME_H_
+
+#include <zephyr/types.h>
+#include <zephyr/drivers/rtc.h>
+
+/** Get universal coordinated time timestamp */
+int sys_time_get_timestamp(int64_t *timestamp);
+/** Set universal coordinated time timestamp */
+int sys_time_set_timestamp(const int64_t *timestamp);
+
+/** Get universal coordinated time datetime */
+int sys_time_get_datetime(struct rtc_time *datetime);
```

Comment:
Its the other way around really :)

The RTC API itself is closely coupled to time.h, struct rtc_time is a 1-1 match to struct tm. The reason for this choice was to keep compatibility with POSIX, while not being dependent on it. This is the reason for using it over struct tm in the general API, not because of RTC drivers. We could move the struct out of rtc.h, have it in sys/time.h instead to make this clearer, actually, we should so that :)

Reply:
Oh yeah, I just realized it's using a standard header. It should really be moved to a separate one.

Generally speaking, we want to avoid adding non-standard functions to standard headers.